src/main.c

```diff
@@ -19,17 +19,75 @@ void hello_world(void)
     }
 }
 
-void bye_world(void)
+void dummy(void)
 {
-    char buf[] = "Bye!\r\n";
 
-    while (1)
+}
+
+void test3(void)
+{
+    char buf[] = "0\r\n";
+
+    for (int i = 0; i < 10; i++)
     {
-        uart_write(UART2, buf, 6);
+        buf[0] = i + '0';
+        uart_write(UART2, buf, 3);
         yield();
     }
 }
 
+void recursive(int step)
+{
+    char buf1[] = "enter  \r\n";
+    char buf2[] = "exit  \r\n";
+    char buf[] = "yield\r\n";
+
+    buf1[6] = step + '0';
+    buf2[5] = step + '0';
+    uart_write(UART2, buf1, 9);
```

Comment:
Use `sizeof` or `strlen`. Manually calculating sizes of the buffers is error-prone.